browser/ui/views/text_recognition_dialog_view.cc

```diff
@@ -0,0 +1,122 @@
+      new TextRecognitionDialogView(image), web_contents)
+      ->Show();
+}
+
+}  // namespace brave
+
+TextRecognitionDialogView::TextRecognitionDialogView(const SkBitmap& image) {
+  SetModalType(ui::MODAL_TYPE_CHILD);
+  SetButtons(ui::DIALOG_BUTTON_OK);
+  SetButtonLabel(ui::DIALOG_BUTTON_OK,
+                 brave_l10n::GetLocalizedResourceUTF16String(
+                     IDS_TEXT_RECOG_DIALOG_CLOSE_BUTTON));
+  SetShowCloseButton(false);
+
+  SetLayoutManager(std::make_unique<views::FlexLayout>())
+      ->SetOrientation(views::LayoutOrientation::kVertical)
+      .SetMainAxisAlignment(views::LayoutAlignment::kStart)
+      .SetInteriorMargin(gfx::Insets::TLBR(24, 26, 0, 26));
+
+  header_label_ = AddChildView(std::make_unique<views::Label>());
+  const int size_diff = 14 - views::Label::GetDefaultFontList().GetFontSize();
+  header_label_->SetFontList(
+      views::Label::GetDefaultFontList()
+          .DeriveWithSizeDelta(size_diff)
+          .DeriveWithWeight(gfx::Font::Weight::SEMIBOLD));
+  header_label_->SetHorizontalAlignment(gfx::ALIGN_LEFT);
+  header_label_->SetProperty(views::kMarginsKey,
+                             gfx::Insets::TLBR(0, 0, 10, 0));
+
+  if (image.empty()) {
+    UpdateContents({});
+    return;
+  }
+
+  StartExtractingText(image);
+}
+
+TextRecognitionDialogView::~TextRecognitionDialogView() = default;
+
+void TextRecognitionDialogView::StartExtractingText(const SkBitmap& image) {
+  header_label_->SetText(brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_TEXT_RECOG_DIALOG_HEADER_IN_PROGRESS));
+
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      {base::MayBlock(), base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN},
+      base::BindOnce(&GetTextFromImage, image),
+      base::BindOnce(&TextRecognitionDialogView::OnGetTextFromImage,
+                     weak_factory_.GetWeakPtr()));
+}
+
+void TextRecognitionDialogView::OnGetTextFromImage(
+    const std::vector<std::string>& text) {
+  UpdateContents(text);
+  AdjustWidgetSize();
+}
+
+void TextRecognitionDialogView::UpdateContents(
+    const std::vector<std::string>& text) {
+  if (text.empty()) {
+    header_label_->SetText(brave_l10n::GetLocalizedResourceUTF16String(
+        IDS_TEXT_RECOG_DIALOG_HEADER_FAILED));
+    return;
+  }
+
+  header_label_->SetText(brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_TEXT_RECOG_DIALOG_HEADER_COMPLETE));
+
+  // Treat each string in |text| as a separated line string.
+  const std::string unified_string = base::JoinString(text, "\n");
+  ui::ScopedClipboardWriter(ui::ClipboardBuffer::kCopyPaste)
+      .WriteText(base::UTF8ToUTF16(unified_string));
+
+  auto* scroll_view = AddChildView(std::make_unique<views::ScrollView>());
+  scroll_view->SetProperty(views::kMarginsKey, gfx::Insets::VH(0, 10));
+  scroll_view->ClipHeightTo(0, 350);
+  auto* label = scroll_view->SetContents(
+      std::make_unique<views::Label>(base::UTF8ToUTF16(unified_string)));
```

Comment:
`base::UTF8ToUTF16(unified_string)` is here and few lines above (just put into var and use in both places).